patches/tModLoader/Terraria/ModLoader/ResourcesDisplayStyleLoader.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Terraria.GameContent.UI;
+
+namespace Terraria.ModLoader
+{
+	public class ResourcesDisplayStyleLoader
+	{
+		// There are no display set IDs, they're stored in their own list
+		// There are 3 of them
+		public const int VanillaDisplaySetCount = 4;
+
+		public static int DisplaySetCount => VanillaDisplaySetCount + displayStyles.Count;
+		internal static readonly List<ModResourcesDisplayStyle> displayStyles = new List<ModResourcesDisplayStyle>();
+
+		internal static int Add(ModResourcesDisplayStyle displayStyle) {
+			if (ModNet.AllowVanillaClients)
+				throw new Exception("Adding display sets breaks vanilla client compatibility");
+
+			displayStyles.Add(displayStyle);
+			Main.PlayerResourcesSets.Add(displayStyle.SetName, displayStyle);
+
+			// Check to see if the requested set has been registered yet
+			// If it has, switch back to it
+			GotoSavedDisplaySet();
```

Comment:
I see no reason for the bottom line to be called here and not after all iloadables are registered.